core/src/test/java/org/elasticsearch/search/geo/GeoShapeIntegrationIT.java

```diff
@@ -479,6 +476,40 @@ public void testOrientationPersistence() throws Exception {
         assertThat(orientation, equalTo(ShapeBuilder.Orientation.CCW));
     }
 
+    @Test
+    public void testPointsOnly() throws Exception {
+        String mapping = XContentFactory.jsonBuilder().startObject().startObject("type1")
+                .startObject("properties").startObject("location")
+                .field("type", "geo_shape")
+                .field("tree", randomBoolean() ? "quadtree" : "geohash")
+                .field("tree_levels", "6")
+                .field("distance_error_pct", "0.01")
+                .field("points_only", true)
+                .endObject().endObject()
+                .endObject().endObject().string();
+
+        assertAcked(prepareCreate("geo_points_only").addMapping("type1", mapping));
+        ensureGreen();
+
+        // test exception handling by attempting to insert a random polygon
+        ShapeBuilder shape = RandomShapeGenerator.createShape(random());
```

Comment:
Good point. The comment should be more clear (and probably moved inside the catch block).  ```RandomShapeGenerator.createShape``` generates a random GeoJson compliant object, so that could be one of [Point, MultiPoint, Linestring, MultiLinestring, Polygon, Multipolygon, GeometryCollection].  If the ```geo_shape``` mapping is defined with ```points_only : true``` then anything other that ```point``` should throw an exception.  That's what we're testing here in the catch block.